packages/build/src/plugins/child/mutations.js

```diff
@@ -61,90 +62,38 @@ const removeProxies = function (value) {
   return originalValue
 }
 
-const addProxy = function (value, keys, { state, configMutations, event }) {
+const addProxy = function (value, keys, { configMutations, event }) {
   // eslint-disable-next-line fp/no-proxy
-  const proxy = new Proxy(value, getReadonlyProxyHandlers({ keys, state, configMutations, event }))
-  PROXIES.set(proxy, value)
-  return proxy
-}
-
-// Retrieve the proxy validating function for each property
-const getReadonlyProxyHandlers = function ({ keys, state, configMutations, event }) {
-  return {
+  const proxy = new Proxy(value, {
     preventExtensions: forbidMethod.bind(undefined, keys, 'validateAnyProperty'),
     setPrototypeOf: forbidMethod.bind(undefined, keys, 'setPrototypeOf'),
     deleteProperty: forbidDelete.bind(undefined, keys),
-    set: validateSet.bind(undefined, { parentKeys: keys, state, configMutations, event }),
-    defineProperty: validateDefineProperty.bind(undefined, { parentKeys: keys, state, configMutations, event }),
-  }
-}
-
-// Triggered when calling `netlifyConfig.{key} = value`
-// eslint-disable-next-line max-params
-const validateSet = function ({ parentKeys, state, configMutations, event }, proxy, lastKey, value, receiver) {
```

Comment:
The `set` handler also trigger the `defineProperty`, so there is no need to define both.